### ISSUE TITLE
shell.openExternal cannot open URL with multiple # on Mac

### ISSUE THREAD

**Issue by chrisyip** [atom/electron#2768]
Our project needs to capture all the click event of `<a>` and decide whether should open the URL in external browser. Code looks like this:

```js
mainWindow.webContents.on('new-window', function (event, url, name) {
  if (shouldOpenExternal(name)) {
    event.preventDefault();
    shell.openExternal(url);
  } else {
    var win = getWindow(name);
    win.loadUrl(url);
    win.show();
  }
});
```

On Windows 10, it works as what we expected. However, on Mac, it doesn't work at all.

Here're screencasts:

![win](https://cloud.githubusercontent.com/assets/189559/9837165/10211af0-5a69-11e5-9b34-030cec897f20.gif)

![mac](https://cloud.githubusercontent.com/assets/189559/9837187/964bf00a-5a69-11e5-9030-ecdbe196f3cc.gif)

Tested Electron `0.32.2` on OS X 10.10, OS X 10.11 GM and Windows 10.

**Comment by hankbao** [comment]
Same issue here.

**Comment by hokein** [comment]
Looks like the test url(http://test.app.dev/.../..d7##) is not a well-formed url. You can see the result returned by `shell.openExternal(your-test-url)` is `false` on OS X. 

The `NSURL`(which is used in `openExternal`) is required a well-formed url as stated in [Apple docs](https://developer.apple.com/library/mac/documentation/Cocoa/Reference/Foundation/Classes/NSURL_Class/#//apple_ref/occ/clm/NSURL/URLWithString:):

> The URL string with which to initialize the NSURL object. Must be a URL that conforms to RFC 2396. This method parses URLString according to RFCs 1738 and 1808.

It probably cannot be fixed in electron due to the requirement of `NSURL`.

**Comment by chrisyip** [comment]
@hokein do you have any workaround?

**Comment by hokein** [comment]
@chrisyip Unfortunately, no much idea about it. You may change to use a well-formed url.

\cc @zcbenz who may have an idea.

**Comment by zcbenz** [comment]
The `open` command works with such URLs, not sure how it made that.

**Comment by zcbenz** [comment]
Fixed by #3849.
